kiwi_lint/similar_string.py

```diff
@@ -0,0 +1,73 @@
+import string
+
+import astroid
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import IAstroidChecker
+from textdistance import levenshtein
+
+
+class SimilarStringChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+    name = "similar-string-checker"
+
+    msgs = {
+        "R5011": (
+            'A similar string to "%s" exists: "%s"',
+            "similar-string",
+            "Similar strings should be avoided for single translation",
+        )
+    }
+
+    def __init__(self, linter):
+        super().__init__(linter)
+        self._dict_of_strings = {}
+
+    @staticmethod
+    def clean_string(text):
+        for word in text:
+            if word not in string.punctuation:
+                text += "".join(word)
+        return text
```

Comment:
Please explain the idea behind this method? Why do we need to clean up the strings ? Isn't the Levenshtein algorithm going to produce similar results if the strings aren't cleaned ?

Reply:
I created this method to clean the punctuations in strings. Without using this function there are many false-positive results which includes 
```
tcms/signals.py:222:16: R5011: A similar string to "NEW: Bug #%(pk)d - %(summary)s" exists: "NEW: TestRun #%(pk)d - %(summary)s" (similar-string)
```
here the operators used in this string is adding up to the similarity of the two string which is not required to be detected.

Reply:
Got it. Good catch. However put this explanation into the doc-string for the method because this sound important.